packages/hardhat/contracts/CCExampleClient.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {ICCGatewayClient} from "./interfaces/ICCGatewayClient.sol";
+import {ICCGateway} from "./interfaces/ICCGateway.sol";
+
+contract CCExampleClient is ICCGatewayClient {
```

Comment:
CCExampleClient didn't call registerRequest()? request() method will fail?

Reply:
The gateway is intended to register the available requests outside the smart-contracts so the logic could be abstracted from them.

I've created another PR which is a gateway-toolkit [tasks, scripts, helpers, ...] to streamline all this process as easier as possible for anybody.